src/shared/parameter_v2/scripts/config_yaml_loader.py

```diff
@@ -0,0 +1,297 @@
+                        )
+
+            if PARAMETER_KEY in metadata:
+
+                # check schema
+                try:
+                    jsonschema.validate(metadata[PARAMETER_KEY], PARAM_DEF_SCHEMA)
+                except jsonschema.exceptions.ValidationError as jsval:
+                    raise ConfigYamlSchemaViolation(
+                        "Schema violation in {}: {}".format(config_file, jsval)
+                    ) from None
+
+                # Get all parameter names as a list, the parameter type comes
+                # first and the name follows in the dictionary. If the schema
+                # check above succeeded, its safe to assume the "name" key
+                # exists in the parameter dictionary
+                param_names = [
+                    list(param_entry.values())[0]["name"]
+                    for param_entry in metadata[PARAMETER_KEY]
+                ]
+
+                # check duplicates
+                if len(param_names) > len(set(param_names)):
+                    raise ConfigYamlMalformed(
+                        "Duplicate parameter detected in {}".format(config_file)
+                    )
+
+                # This is an ugly artifact of how the yaml is defined and loaded
+                # We are extracting all the requested types to check that
+                # they are all supported. This is the one thing the schema
+                # can't validate that we would like to check
+                requested_types = [
+                    key[0]
+                    for key in [list(entry.keys()) for entry in metadata[PARAMETER_KEY]]
+                ]
+
+                # check if type requested is supported
+                for requested_type in requested_types:
+                    if requested_type not in SUPPORTED_TYPES:
+                        raise ConfigYamlMalformed(
+                            "{} type unsupported in {}".format(
+                                requested_type, config_file
+                            )
+                        )
+
+    @staticmethod
+    def __detect_cycles_in_config_metadata(config_metadata: dict):
+        """Creates a DiGraph from all the included configs and checks if there
+        are cycles. Raises to the main thread if a cycle is detected
+
+        :raises: ConfigYamlCycleDetected
+        :param config_metadata: Metadata describing params and config includes
+        :type config_metadata: dict
+
+        """
+        edges = []
+
+        for config, metadata in config_metadata.items():
+            if INCLUDE_KEY in metadata:
+                for included_config in metadata[INCLUDE_KEY]:
+                    edges.append((config, included_config))
+
+        G = networkx.DiGraph(edges)
+
+        for cycle in networkx.simple_cycles(G):
+            raise ConfigYamlCycleDetected(
+                "Cycle detected in the include statements: "
+                + " -> ".join(cycle + [cycle[0]])
+            )
```

Comment:
will this actually print out all the cycles or will it exit on the first cycle?

Reply:
it will exit on the first cycle, you would re-run after fixing it to detect additional cycles